source/hic_hal/freescale/kl26z/uart.c

```diff
@@ -283,9 +237,18 @@ void UART_RX_TX_IRQHandler(void)
         if ((s1 & UART_S1_NF_MASK) || (s1 & UART_S1_FE_MASK)) {
             errorData = UART->D;
         } else {
-            read_buffer.data[read_buffer.idx_in++] = UART->D;
-            read_buffer.idx_in &= (UART_BUFFER_SIZE - 1);
-            read_buffer.cnt_in++;
+            uint32_t free;
+            uint8_t data;
+            
+            data = UART1->D;
+            free = circ_buf_count_free(&read_buffer);
+            if (free > RX_OVRF_MSG_SIZE) {
+                circ_buf_push(&read_buffer, data);
+            } else if (RX_OVRF_MSG_SIZE == free) {
+                circ_buf_write(&read_buffer, (uint8_t*)RX_OVRF_MSG, RX_OVRF_MSG_SIZE);
```

Comment:
@flit do you think this behavior is worse that just dropping the data? Either way this seems likely to cause problems with some protocols. With an error message in the serial stream at least the cause of serial data loss can be identified. I'm open to suggestions on alternate ways to fix this.

One important case this helps with is in rare cases in mbed-os testing there are dropped segments of test output that cause the testing to indicate failure. It has been hard to determine if the failure was due to DAPLink dropping data due to an overflow or a real failure.